ReactNative/ReactTwitter/core/views/mixins/android-back-navigation.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { BackAndroid, Navigator } from 'react-native'
+
+/*eslint-disable no-unused-vars*/
+var AndroidBackNavigationMixin = {
+  propTypes: {
+    navigator: React.PropTypes.instanceOf(Navigator).isRequired
+  },
+
+  componentDidMount () {
+    BackAndroid.addEventListener('hardwareBackPress', () => {
+      if (this.props.navigator.getCurrentRoutes().length === 1) {
+        return false
+      }
+      this.props.navigator.pop()
+      return true
+    })
+  },
+
+  componentWillUnmount: function () {
+    BackAndroid.removeEventListener('hardwareBackPress')
```

Comment:
Shouldn't you provide also a reference to the `handler` that you added previously?